src/harness/testcases/WINNF_FT_S_PCR_testcase.py

```diff
@@ -0,0 +1,178 @@
+    ppa_record_2, pal_record_2 = makePpaAndPalRecordsConsistent(ppa_record_2,
+                                                            [pal_record_2],
+                                                            pal_low_frequency,
+                                                            pal_high_frequency,
+                                                            device_a['userId'])
+    device_b['installationParam']['latitude'], device_b['installationParam'][
+               'longitude'] = getRandomLatLongInPolygon(ppa_record_2)
+    device_b['userId'] = pal_record_2['licensee']
+
+    config = {
+        'device_a': device_a,
+        'device_b': device_b,
+        'pal_record_1': pal_record_1,
+        'pal_record_2': pal_record_2
+
+    }
+    writeConfig(filename, config)
+
+  @configurable_testcase(generate_PCR_1_default_config)
+  def test_WINNF_FT_S_PCR_1(self, config_filename):
+    """ Successful Maximum PPA Creation
+
+    Checks PPA generated by SAS UUT shall be fully contained within the service area.
+    """
+    # Load the Config file
+    config = loadConfig(config_filename)
+    device_a = config['device_a']
+    device_b = config['device_b']
+    pal_record_1 = config['pal_record_1']
+    pal_record_2 = config['pal_record_2']
+
+    # Step 1: Inject one or more PAL records
+    self._sas_admin.InjectPalDatabaseRecord(pal_record_1[0])
+    self._sas_admin.InjectPalDatabaseRecord(pal_record_2[0])
+
+    # Step 2: Register device_a in SAS UUT
+    self._sas_admin.InjectFccId({'fccId': device_a['fccId']})
+    self._sas_admin.InjectUserId({'userId': device_a['userId']})
+    self._sas_admin.InjectFccId({'fccId': device_b['fccId']})
+    self._sas_admin.InjectUserId({'userId': device_b['userId']})
+
+    # Do registeration and check registration response of device_a and device_b in SAS UUT
+    cbsd_ids = self.assertRegistered([device_a, device_b])
+
+    # Step 3: Triggers SAS UUT to create a PPA boundary
+    palIds = [pal_record_1['palId']]
+    ppaIds = self._sas_admin.TriggerPpaCreation(cbsd_ids, palIds)
+
+    self.assertGreaterEqual(len(ppaIds),0)
+    # Step 4: Triggers the SAS UUT to generate the Full Activity Dump
+    self._sas_admin.TriggerCreateActivityDump()
+
+    # Step 5: SAS Test Harness uses the SAS-SAS protocol to retrieve the PPA zone
+    full_activity_dump = self._sas.GetSasDumpRecord()
+    ppa_record_url = None
+    for dump_file in full_activity_dump['files']:
+      if dump_file['recordType'] == 'zone':
+        ppa_record_url = dump_file['url']
+        break
+
+    # Checking whether SAS dump data contains the ppa zone url
+    self.assertIsNotNone(ppa_record_url)
+    self.polygon_dump = self._sas.GetSasPPADumpRecord(ppa_record_url)
+
+    # Step 6: Using PPA Creation Reference Model to calculate the maximum PPA
+    devices_list = [device_a, device_b]
+    pal_record_list = [pal_record_1]
+    pal_low_frequency = pal_record_1['channelAssign']['primaryAssignmen']['lowFrequency']
```

Comment:
These are not valid keys for the `pal_record_1` dictionary, so this code will not even be able to execute.